cmd/wallet/import.go

```diff
@@ -1,80 +1,132 @@
-		}
-
-		var kindRaw string
-		err = survey.AskOne(&survey.Select{
-			Message: "Kind:",
-			Options: supportedKinds,
-		}, &kindRaw)
-		cobra.CheckErr(err)
-
-		var kind wallet.ImportKind
-		err = kind.UnmarshalText([]byte(kindRaw))
-		cobra.CheckErr(err)
-
-		// Ask for wallet configuration.
-		afCfg, err := af.GetConfigFromSurvey(&kind)
-		cobra.CheckErr(err)
-
-		// Ask for import data.
-		var answers struct {
-			Data string
-		}
-		questions := []*survey.Question{
-			{
-				Name:     "data",
-				Prompt:   af.DataPrompt(kind, afCfg),
-				Validate: af.DataValidator(kind, afCfg),
-			},
-		}
-		err = survey.Ask(questions, &answers)
-		cobra.CheckErr(err)
-
-		// Ask for passphrase.
-		passphrase := common.AskNewPassphrase()
-
-		accCfg := &config.Account{
-			Kind:   af.Kind(),
-			Config: afCfg,
-		}
-		src := &wallet.ImportSource{
-			Kind: kind,
-			Data: answers.Data,
-		}
-
-		err = cfg.Wallet.Import(name, passphrase, accCfg, src)
-		cobra.CheckErr(err)
-
-		err = cfg.Save()
-		cobra.CheckErr(err)
-	},
+var (
+	algorithm string
+	number    uint32
+	secret    string
+
+	passphrase string
+	accCfg     *config.Account
+	kind       wallet.ImportKind
+
+	importCmd = &cobra.Command{
+		Use:   "import <name> [flags]",
+		Short: "Import an existing account",
+		Args:  cobra.ExactArgs(1),
+		Run: func(cmd *cobra.Command, args []string) {
+			cfg := config.Global()
+			name := args[0]
+
+			checkAccountExists(cfg, name)
+
+			// NOTE: We only support importing into the file-based wallet for now.
+			af, err := wallet.Load(walletFile.Kind)
+			cobra.CheckErr(err)
+
+			afCfg := make(map[string]interface{})
+
+			if !common.GetAnswerYes() {
+				// Ask for import kind.
+				var supportedKinds []string
+				for _, kind := range af.SupportedImportKinds() {
+					supportedKinds = append(supportedKinds, string(kind))
+				}
+
+				var kindRaw string
+				err = survey.AskOne(&survey.Select{
+					Message: "Kind:",
+					Options: supportedKinds,
+				}, &kindRaw)
+				cobra.CheckErr(err)
+
+				err = kind.UnmarshalText([]byte(kindRaw))
+				cobra.CheckErr(err)
+
+				// Ask for wallet configuration.
+				afCfg, err = af.GetConfigFromSurvey(&kind)
+				cobra.CheckErr(err)
+
+				// Ask for import data.
+				var answers struct {
+					Data string
+				}
+				questions := []*survey.Question{
+					{
+						Name:     "data",
+						Prompt:   af.DataPrompt(kind, afCfg),
+						Validate: af.DataValidator(kind, afCfg),
+					},
+				}
+				err = survey.Ask(questions, &answers)
+				cobra.CheckErr(err)
+
+				// Ask for passphrase.
+				passphrase = common.AskNewPassphrase()
+			}
+
+			if algorithm != "" {
+				afCfg["algorithm"] = algorithm
+			}
+
+			if number != 0 {
+				afCfg["number"] = 0
+			}
+
+			accCfg = &config.Account{
+				Kind:   af.Kind(),
+				Config: afCfg,
+			}
+
+			whitespace := strings.Contains(secret, " ")
+			if whitespace {
+				kind = wallet.ImportKindMnemonic
+			} else {
+				kind = wallet.ImportKindPrivateKey
+			}
```

Comment:
I would set `kind` based on `algorithm` instead of doing some magic autodetection.

```suggestion
			if algorithm==wallet.AlgorithmEd25519Raw || algorithm==wallet.AlgorithmSecp256k1Raw || algorithm==wallet.AlgorithmSr25519Raw {
				kind = wallet.ImportKindPrivateKey
			} else {
				kind = wallet.ImportKindMnemonic
			}
```